integration_tests/src/main/python/conftest.py

```diff
@@ -238,16 +239,18 @@ def spark_tmp_path(request):
     ret = request.config.getoption('tmp_path')
     if ret is None:
         ret = '/tmp/pyspark_tests/'
-    ret = ret + '/' + str(random.randint(0, 1000000)) + '/'
+    pid_dir = f'{ret}/{os.getpid()}'
```

Comment:
I think if we also add worker_id / PYTEST_XDIST_WORKER when defined it should resolve collisions. 

I wonder why we don't use `integration_tests/target/tmp` path prefix?

Reply:
> I think if we also add worker_id / PYTEST_XDIST_WORKER when defined it should resolve collisions.

Is that going to get us any more unique than we're already getting via the process ID?

> I wonder why we don't use integration_tests/target/tmp path prefix?

Good question.  May be because this isn't always done as part of a build.  I can file a followup issue, since changing that is not really in the scope of this PR.

Reply:
>>  I think if we also add worker_id / PYTEST_XDIST_WORKER when defined it should resolve collisions.
> Is that going to get us any more unique than we're already getting via the process ID?

I am not sure, but it allows easier mapping to the info logged by xdist and is consistent with _spark_init https://github.com/NVIDIA/spark-rapids/blob/1e851c24139f688040225c74ed73abb8352bd3bd/integration_tests/src/main/python/spark_init_internal.py#L42

Reply:
I think it depends on the mode of running the tests. Some of the CI jobs are running with cluster mode, so you'd need the directory `integration_tests/target/tmp` to exist in the AM container, so the `/tmp/pyspark_tests` in this case, could be an HDFS directory instead of one in the local file system.

Reply:
oh I see, then I would also include the host name in the path prefix

Reply:
I've added the hostname and the xdist worker_id in the temp directory name.